darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterStrategyFactoryImpl.java

```diff
@@ -63,13 +71,19 @@
   private final Random _random;
   private final LoadBalancerClusterListener _clusterListener;
   private final DarkClusterVerifierManager _verifierManager;
+  private final ScheduledExecutorService _rateLimiterScheduler;
+  private final ExecutorService _rateLimiterExecutor;
+  private final ExpiringCircularBuffer<Callback<None>> _rateLimiterStoredCallbacks;
 
   public DarkClusterStrategyFactoryImpl(@Nonnull Facilities facilities,
                                         @Nonnull String sourceClusterName,
                                         @Nonnull DarkClusterDispatcher darkClusterDispatcher,
                                         @Nonnull Notifier notifier,
                                         @Nonnull Random random,
-                                        @Nonnull DarkClusterVerifierManager verifierManager)
+                                        @Nonnull DarkClusterVerifierManager verifierManager,
+                                        @Nonnull ScheduledExecutorService rateLimiterScheduler,
+                                        @Nonnull ExecutorService rateLimiterExecutor,
+                                        @Nonnull ExpiringCircularBuffer<Callback<None>> rateLimiterStoredCallbacks)
```

Comment:
Instead of passing all three parameters to construct a RateLimiter, can we simply pass in a RateLimiter instance instead?

Reply:
It's passed this way in order to give access to setCapacity and setTtl on _rateLimiterStoredCallbacks. Could probably pass just the GuaranteedRateLimiter if I added public methods to GuaranteedRateLimiter that forward to the underlying setCapacity and setTtl methods on GuaranteedRateLimiter's ExpiringCircularBuffer. I'll draft that up